x/mongo/driver/batches.go

```diff
@@ -61,6 +63,12 @@ func (b *Batches) AdvanceBatch(maxCount, targetBatchSize int) error {
 		splitAfter++
 	}
 
+	// if there are no documents, take the first one.
+	// this can happen if there is a document that is smaller than maxDocSize but greater than targetBatchSize.
+	if splitAfter == 0 {
```

Comment:
This is the relevant code for putting a document that's larger than 2MiB into its own batch. The code above will validate the the document is the correct size (below `maxDocSize`) but won't add it to the batch because that would exceed `targetBatchSize`. In this case, we unconditionally take the last document by setting `splitAfter` to 1.

Reply:
Ack